test/e2e/util/install.go

```diff
@@ -0,0 +1,234 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type Cleanup func(ctx context.Context) error
+
+// CleanupSentinel is a helper for returning a no-op cleanup function.
+func CleanupSentinel(_ context.Context) error {
+	return nil
+}
+
+// SkippedCleanupSteps parses $SKIP_CLEANUP as a comma-delimited list of cleanup steps to skip.
+func SkippedCleanupSteps() sets.Set[string] {
+	skip := os.Getenv("SKIP_CLEANUP")
+	if skip == "" {
+		return sets.New[string]()
+	}
+	parts := strings.Split(skip, ",")
+	return sets.New[string](parts...)
+}
+
+type Builder func(ctx context.Context, logger logr.Logger, opts *Options) (Cleanup, error)
+
+// InstallHyperShiftCRDs generates and applies CRDs to the cluster for setup of the HyperShift Operator.
+//
+// A closure is returned that knows how to clean up the content on the cluster.
+func InstallHyperShiftCRDs(ctx context.Context, logger logr.Logger, opts *Options) (Cleanup, error) {
+	installLogPath := filepath.Join("install", "hypershift-install-crds.log")
+	renderCmd := exec.CommandContext(ctx, opts.HyperShiftCLIPath,
```

Comment:
why are we running the CLI command locally instead of just calling the CLI code directly?

Reply:
this code is surfaced from the integration package. I'd say any improvement needed there can be done in a separate PR as long as there're no huge concerns.

The reason it's copied is after a discussion with @enxebre where we agreed to copy any code we'll use from there as a signal to what exactly is useful for the current e2e from that integration package and be able to track that.

Reply:
my question is how is it better to exec a command locally, instead of calling our own code directly?

Reply:
> Basically, using this code as-is allows us to get a running upgrade test in place faster, and we can later make it better.

It's not really `as is` if we are introducing new code (even if its copied from somewhere else)

Reply:
Addressed